presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -175,7 +158,7 @@ void PrestoServer::run() {
     httpExecThreads = systemConfig->httpExecThreads();
     environment_ = nodeConfig->nodeEnvironment();
     nodeId_ = nodeConfig->nodeId();
-    address_ = nodeConfig->nodeIp(getLocalIp);
+    address_ = nodeConfig->nodeIp(std::bind(&PrestoServer::getLocalIp, this));
```

Comment:
Why not just getLocalIp() ?

Reply:
We need to make `getLocalIp` a member function to provide process specific IP. Since it's a member function, we need to pass the fully qualified function name and the object instance for lambda argument. Just passing `getLocalIp` for instance function won't compile.